### PR TITLE
Delete taxonomy UI

### PR DESCRIPTION
**Pull request by allisonking**
Closes https://github.com/ethyca/fides/issues/859

### Code Changes

* [x] Add delete calls to each taxonomy slice
* [x] Wire up delete button with confirmation modal
* [x] Only show delete button on nodes without children (related to https://github.com/ethyca/fides/issues/1005)
* [x] Cypress tests for deleting taxonomy fields

### Steps to Confirm

* [ ] Go to `/taxonomy` and try deleting some fields

### Pre-Merge Checklist

* [x] All CI Pipelines Succeeded
* Documentation Updated:
  - [ ] documentation complete, or draft/outline provided (tag docs-team to complete/review on this branch)
  - [ ] documentation issue created (tag docs-team to complete issue separately)
* [x] Issue Requirements are Met
* [x] Relevant Follow-Up Issues Created
  * https://github.com/ethyca/fides/issues/1005
* [x] Update `CHANGELOG.md`

### Description Of Changes

https://user-images.githubusercontent.com/24641006/186188681-8f9ef046-b3d8-4bce-9489-ce2481cb879c.mov

Error state

https://user-images.githubusercontent.com/24641006/186188950-3f03be3d-bc8e-4cf2-b46f-fafaaa8aa2d1.mov

Parent nodes do not render the delete button
image

### CODE REVIEW

File: clients/ctl/admin-ui/src/features/common/AccordionTree.tsx [hunk 1]
```diff
@@ -91,7 +110,11 @@ const AccordionTree = ({ nodes, onEdit, focusedKey }: Props) => {
             {node.label}
           </AccordionButton>
           {isHovered ? (
-            <ActionButtons node={hoverNode} onEdit={onEdit} />
+            <ActionButtons
+              node={hoverNode}
+              onEdit={onEdit}
+              onDelete={onDelete}
```
Comment (ssangervasi) [inline clients/ctl/admin-ui/src/features/common/AccordionTree.tsx:116]: The AccordionTree is nice and generic, other than this pass-through of the action callbacks. We could replace those callback props with a react child prop and then the consumer could decide what's shown on hover. Something like:
```ts
interface Props {
  nodes: TreeNode[];
  focusedKey?: string;
  renderHover?: (node: TreeNode) => React.ReactNode
}
const AccordionTree = ({ ... }: Props) => { ... }

// Usage
 <AccordionTree
  nodes={taxonomyNodes}
  focusedKey={editEntity?.fides_key}
  renderHover={(node) => <ActionButtons node={node} onEdit... } />
```

File: clients/ctl/admin-ui/src/features/taxonomy/TaxonomyTabContent.tsx [hunk 2]
```diff
@@ -1,7 +1,19 @@
...
+  useToast,
+} from "@fidesui/react";
 import { useMemo, useState } from "react";
 
+import ConfirmationModal from "~/features/common/ConfirmationModal";
+import { isErrorResult } from "~/types/errors";
+
 import AccordionTree from "../common/AccordionTree";
+import { getErrorMessage } from "../common/helpers";
+import { errorToastParams, successToastParams } from "../common/toast";
```
Comment (ssangervasi) [inline clients/ctl/admin-ui/src/features/taxonomy/TaxonomyTabContent.tsx:16]: Nit: mix of `../common` and `~/features/common`

File: clients/ctl/admin-ui/src/features/taxonomy/TaxonomyTabContent.tsx [hunk 3]
```diff
@@ -40,29 +61,66 @@ const TaxonomyTabContent = ({ useTaxonomy }: Props) => {
...
+        <AccordionTree
+          nodes={taxonomyNodes}
+          onEdit={handleSetEditEntity}
+          onDelete={handleSetDeleteKey}
+          focusedKey={editEntity?.fides_key}
         />
-      ) : null}
-    </SimpleGrid>
+        {editEntity ? (
+          <TaxonomyFormBase
```
Comment (ssangervasi) [inline clients/ctl/admin-ui/src/features/taxonomy/TaxonomyTabContent.tsx:98]: I noticed that if I click "Edit" to show the form for a taxonomy item, then hit "Delete" on that same item (and confirm) the edit form will stick around for the deleted item. Attempting to "Update" the item will then error, of course. So maybe the call to `handleSetDeleteKey` needs to also clear the EditEntity.

File: clients/ctl/admin-ui/src/features/taxonomy/hooks.tsx [hunk 4]
```diff
@@ -96,9 +103,10 @@ export const useDataUse = (): TaxonomyHookData<DataUse> => {
       "Legitimate interest impact assessment",
   };
 
-  const [edit] = useUpdateDataUseMutation();
+  const [onEdit] = useUpdateDataUseMutation();
+  const [onDelete] = useDeleteDataUseMutation();
```
Comment (ssangervasi) [inline clients/ctl/admin-ui/src/features/taxonomy/hooks.tsx:107]: Nit: the "on" prefix doesn't read quite right here. The mutation isn't an event handler, it is the actual  "update" or "delete" operation.

**Review by ssangervasi — COMMENTED** [review]
At first the title "Delete taxonomy UI" made me think the UI itself was going away 😆 

A few comments. The edit form sticking around is the only problem.

File: clients/ctl/admin-ui/src/features/common/AccordionTree.tsx (hunk 1 above)
Comment (allisonking) [reply clients/ctl/admin-ui/src/features/common/AccordionTree.tsx:116]: ah! yeah I had been bothered by that, but didn't know a good pattern to work around it. thanks for the great suggestion 👍

File: clients/ctl/admin-ui/src/features/taxonomy/hooks.tsx (hunk 4 above)
Comment (allisonking) [reply clients/ctl/admin-ui/src/features/taxonomy/hooks.tsx:107]: ah yeah good point... I originally had `delete` but that conflicted with the default `delete` so I mindlessly changed it to `onDelete` 😅 I'll find a better word!

**Comment by allisonking** [comment]
Thanks @ssangervasi ! Addressed all your comments, should be ready for review again 👍 

Note that I merged https://github.com/ethyca/fides/pull/990 this morning, which prevents deleting default taxonomy fields. So our days of deleting those are over, and we'll get an error right now in the UI. https://github.com/ethyca/fides/issues/858 will fix that and not render the delete button on default fields

**Review by ssangervasi — APPROVED** [review]

**Pull request merged**